src/linker/pipeline_schema.py

```diff
@@ -10,7 +11,7 @@ class PipelineSchema:
     def __init__(self, name, validate_input) -> None:
         self.name = name
         self.validate_input: Callable = validate_input
-        self.steps = []
+        self.steps: List(Step) = []
```

Comment:
I think the proper type hint uses square brackets? `List[Step]`. That said, I don't really run static type checkers as part of my workflow...I'm sure there are plenty of mistakes.